graphene/validation/disable_introspection.py

```diff
@@ -0,0 +1,19 @@
+from graphql import GraphQLError
+from graphql.language import FieldNode
+from graphql.validation import ValidationRule
+
+from ..utils.is_introspection_key import is_introspection_key
+
+
+class DisableIntrospection(ValidationRule):
+    def enter_field(self, node: FieldNode, *_args):
+        field_name = node.name.value
+        if not is_introspection_key(field_name):
+            return
+
+        self.report_error(
+            GraphQLError(
+                f"Cannot query '{field_name}': introspection is disabled.",
+                node,
+            )
+        )
```

Comment:
I think is better the following logic, as it's a bit simpler and easier to follow:
```suggestion
        if is_introspection_key(field_name):
            self.report_error(
                GraphQLError(
                    f"Cannot query '{field_name}': introspection is disabled.",
                    node,
                )
            )
```